FWCore/ParameterSet/interface/PluginDescription.h

```diff
@@ -200,9 +200,10 @@ namespace edm {
   private:
     std::string findType(edm::ParameterSet const& iPSet) const {
       if (typeLabelIsTracked_) {
-        if (iPSet.existsAs<std::string>(typeLabel_) || defaultType_.empty()) {
+        CMS_SA_ALLOW if (iPSet.existsAs<std::string>(typeLabel_) || defaultType_.empty()) {
```

Comment:
Just have a question for my understanding. Do you put CMS_SA_ALLOW here not in the next line of return by intension? Will it give the same results?

Reply:
> Do you put CMS_SA_ALLOW here not in the next line of `return` by intension?

Yes

> Will it give the same results?

No. The `CMS_SA_ALLOW` macro expands to `[[cms::sa_allow]]` attribute (for select compilers), that the clang-based static analyzer then uses to ignore warnings. The attribute needs to "bound" to the conditional expression inside `if (...)`, because that's where the `existsAs()` call is. This is my best guess how it could work, but I'm not 100 % sure.

In the other case the `existsAs()` is part of the `return` statement.